homeassistant/components/spider/climate.py

```diff
@@ -109,7 +107,12 @@ def hvac_mode(self):
     @property
     def hvac_modes(self):
         """Return the list of available operation modes."""
-        return SUPPORT_HVAC
+        hvac_mode_list = []
+        for operation_value in self.support_hvac:
+            hvac_mode_list.append(
+                SPIDER_STATE_TO_HA.get(operation_value, "Invalid operation value")
```

Comment:
https://github.com/home-assistant/core/pull/44575#discussion_r550924174 is still not addressed.

Reply:
Done

Reply:
It is definitely not done.
The concern was that `hvac_modes` must return a list which only contains values found in `HVAC_MODES`, i.e. a subset of:
https://github.com/home-assistant/core/blob/625bbe623812ca2ea6b45371157b9bc2e33727b6/homeassistant/components/climate/const.py#L25-L33
As the PR is now, `"Invalid operation value"` could be in the list returned by `hvac_modes`, which is not in line with the requirements set out by the parent `ClimateEntity` class.
Moving the list generation from `__init__` to `hvac_modes` does not change the fact that the list could contain values which are not allowed. It only causes the list to be recreated every time `hvac_modes` is called, which is unnecessary.
My suggestions from before still stand:
* Generate the modes list in `__init__` and just return it from `hvac_modes`
* Do not include any invalid values in the list. Omit all invalid values and log a warning instead.

Reply:
Sorry, forgot to push the changes.

Commit:
https://github.com/home-assistant/core/pull/44575/commits/6835da58f562d344aca6df3d3569420d76afd8b1